platform-common-pdf/src/main/java/com/softicar/platform/common/pdf/PdfMerger.java

```diff
@@ -0,0 +1,51 @@
+package com.softicar.platform.common.pdf;
+
+import com.softicar.platform.common.io.stream.InputStreamCollection;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.function.Supplier;
+import org.apache.pdfbox.io.MemoryUsageSetting;
+import org.apache.pdfbox.multipdf.PDFMergerUtility;
+import org.apache.pdfbox.multipdf.PDFMergerUtility.DocumentMergeMode;
+
+/**
+ * Merges a {@link Collection} of PDFs into a single PDF.
+ *
+ * @author Oliver Richers
+ */
+public class PdfMerger {
+
+	private final Collection<Supplier<InputStream>> pdfInputStreamSuppliers;
+	private final Supplier<OutputStream> outputStreamSupplier;
+
+	public PdfMerger(Supplier<OutputStream> outputStreamSupplier) {
+
+		this.pdfInputStreamSuppliers = new ArrayList<>();
+		this.outputStreamSupplier = outputStreamSupplier;
```

Comment:
```suggestion
		this.outputStreamSupplier = Objects.requireNonNull(outputStreamSupplier);
```
We should fail early.

Also, we could flip those lines. Same goes for the field declarations.

Reply:
ok